src/app/loadout-builder/utils.ts

```diff
@@ -101,7 +100,7 @@ function getEnergySpendTierBoundaryHash(item: DimItem, tier: UpgradeSpendTier) {
       boundaryHash = UpgradeMaterialHashes.ascendantShard;
       break;
     case UpgradeSpendTier.AscendantShardsNotExotic: {
-      if (!isExotic) {
+      if (!item.isExotic) {
```

Comment:
I don't know why but I had in my head that we didn't use this? Has it come back or did I get it wrong?

Reply:
It's there and it's a bit cleaner. There is a semantic difference - if you want "is this exotic rarity" use isExotic. If you want to enforce rules about two items with the same equippingLabel not being equipped at once (in the loadout or item move code) use equippingLabel. I think all the cases I changed were actually looking for exotics.